Registry/RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/service/CSourceServiceTest.java

```diff
@@ -1,181 +1,401 @@
+			} catch (Exception e) {
+				Assert.assertEquals(registrationid + " not found", e.getMessage());
+			}
+
+		} catch (Exception ex) {
+			Assert.fail();
+		}
+	}
+
+	/**
+	 * this method is use for get CSource Registry
+	 */
+	@Test
+	public void getCSourceRegistrationByIdTest() throws Exception {
+		try {
+
+			ArrayListMultimap<String, String> csourceIds = ArrayListMultimap.create();
+			csourceIds.put(AppConstants.INTERNAL_NULL_KEY, "urn:ngsi-ld:ContextSourceRegistration:csr3");
+			// when(csourceInfoDAO.getAllIds()).thenReturn(csourceIds);
+			HashMap<String, List<String>> result = new HashMap<String, List<String>>();
+			List<String> entities = new ArrayList<String>();
+			result.put(AppConstants.INTERNAL_NULL_KEY, entities);
+			when(csourceInfoDAO.getAllEntities()).thenReturn(result);
+			ReflectionTestUtils.setField(csourceService, "AUTO_REG_STATUS", "active");
+			Method postConstruct = CSourceService.class.getDeclaredMethod("loadStoredEntitiesDetails"); // methodName,parameters
+			postConstruct.setAccessible(true);
+			postConstruct.invoke(csourceService);
+			multimaparr.put("content-type", "application/json");
+			when(csourceInfoDAO.getEntity(any(), any())).thenReturn(payload);
+			String getresult = csourceService.getCSourceRegistrationById(null,
+					"urn:ngsi-ld:ContextSourceRegistration:csr3");
+			Assert.assertEquals(payload, getresult);
+		} catch (Exception ex) {
+			Assert.fail();
+		}
+	}
+
+	/**
+	 * this method is use for get CSource Registry if ID is not exist
+	 */
+	@Test
+	public void getCSourceRegistrationByIdNotExistTest() throws Exception {
+		try {
+
+			ArrayListMultimap<String, String> csourceIds = ArrayListMultimap.create();
+			csourceIds.put(AppConstants.INTERNAL_NULL_KEY, "urn:ngsi-ld:ContextSourceRegistration:csr3");
+			HashMap<String, List<String>> result = new HashMap<String, List<String>>();
+			List<String> entities = new ArrayList<String>();
+			result.put(AppConstants.INTERNAL_NULL_KEY, entities);
+			when(csourceInfoDAO.getAllEntities()).thenReturn(result);
+			ReflectionTestUtils.setField(csourceService, "AUTO_REG_STATUS", "active");
+			Method postConstruct = CSourceService.class.getDeclaredMethod("loadStoredEntitiesDetails");
+			postConstruct.setAccessible(true);
+			postConstruct.invoke(csourceService);
+			multimaparr.put("content-type", "application/json");
+			String registrationid = "urn:ngsi-ld:ContextSourceRegistration:csr3";
+			when(csourceInfoDAO.getEntity(any(), any()))
+					.thenThrow(new ResponseException(ErrorType.NotFound, registrationid + " not found"));
+			try {
+				csourceService.getCSourceRegistrationById(null, "urn:ngsi-ld:ContextSourceRegistration:csr4");
+			} catch (Exception e) {
+				Assert.assertEquals(registrationid + " not found", e.getMessage());
+			}
+
+		} catch (Exception ex) {
+			Assert.fail();
+		}
+	}
+
+	/**
+	 * this method is use for get CSource Registry
+	 */
+	@Test
+	public void getCSourceRegistrationTest() throws Exception {
```

Comment:
[Trival] please change method name as per the test case.

Reply:
done